velox/type/Conversions.h

```diff
@@ -59,11 +59,60 @@ struct Converter {
   }
 };
 
-/// Presto compatible cast of strings to boolean. There is a set of strings
-/// allowed to be casted to boolean. These strings are `t, f, 1, 0, true, false`
-/// and their upper case equivalents. Casting from other strings to boolean
-/// throws.
-Expected<bool> castToBoolean(const char* data, size_t len);
+/// This function casts a string to a boolean value. There is a set of strings
+/// allowed to be casted to boolean. Casting from other strings to boolean
+/// throws. By default, Presto compatible cast is conducted. The allowed strings
+/// are `t, f, 1, 0, true, false` and their upper case equivalents.
+/// @tparam TPolicy The specified cast policy. When set as SparkCastPolicy, the
+/// allowed strings are `t, f, 1, 0, y, n, true, false, yes, no` and their upper
```

Comment:
It might be more readable if the comment describe the differences between Presto and Spark policies.

```
PrestoCastPolicy allows `t, f, 1, 0, true, false` strings and ...

SparkCastPolicy allows all strings that PrestoCastPolicy allows plus `y, n, yes, no` and their ...
```

Reply:
Thanks for the suggestion. I find Presto cast could also use 'LegacyCastPolicy' but its behavior should be the same with 'PrestoCastPolicy' as Presto's implementation in [VarcharOperators](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java#L111-L138) is not controlled by a config (please kindly correct me if I'm wrong). Changed the comment as below.

> The specified cast policy. PrestoCastPolicy and
LegacyCastPolicy allow `t, f, 1, 0, true, false` and their upper case
equivalents. SparkCastPolicy allows all strings that they allow plus `y, n,
yes, no` and their upper case equivalents.

Reply:
I believe LegacyCastPolicy is used by a streaming system at Meta. It uses Presto functions, but requires slightly different behavior in some cases. CC: @bikramSingh91 @kgpai @kagamiori

Reply:
Thanks for the explain. In Velox, Presto cast uses 'LegacyCastPolicy' if the query config 'kLegacyCast' is enabled.

Reply:
@rui-mo Should we document the behavior of this API when TPolicy is LegacyCastPolicy ?

Reply:
It is documented as below, having the same behavior with PrestoCastPolicy. Thanks.
```
/// @tparam TPolicy PrestoCastPolicy and LegacyCastPolicy allow `t, f, 1, 0,
/// true, false` and their upper case equivalents. SparkCastPolicy additionally
/// allows 'y, n, yes, no' and their upper case equivalents.
```

Reply:
Perfect. Thanks.